src/Tempest/View/src/Attributes/ElseAttribute.php

```diff
@@ -4,39 +4,23 @@
 
 namespace Tempest\View\Attributes;
 
-use Exception;
 use Tempest\View\Attribute;
 use Tempest\View\Element;
-use Tempest\View\Elements\EmptyElement;
-use Tempest\View\Elements\GenericElement;
+use Tempest\View\Elements\PhpIfElement;
+use Tempest\View\Exceptions\InvalidElement;
 
 final readonly class ElseAttribute implements Attribute
 {
-    public function apply(Element $element): Element
+    public function apply(Element $element): ?Element
     {
         $previous = $element->getPrevious();
-        $previousCondition = false;
 
-        if (! $previous instanceof GenericElement) {
-            throw new Exception("Invalid preceding element before :else");
+        if (! $previous instanceof PhpIfElement) {
+            throw new InvalidElement('There needs to be an if or elseif element before.');
```

Comment:
TODO: improved exceptions: we need to include the template filename and position — if possible — for easy debugging